### PR TITLE
feat: Add project proto & generate

### PR DESCRIPTION
**Pull request by arsenetar**
Initial version, may have slight addition needed to `AssetParameters` to fully implement.

Also fixed issue with no longer using the provided `protoc` for js (system `protoc` may be a broken version).

### CODE REVIEW

**Review by palp — APPROVED** [review]
Some consideration needs to be given to the way the protoc binary is used, but it's not blocking this from being merged.

File: CMakeLists.txt [hunk 1]
```diff
@@ -114,31 +114,31 @@ if (NOT ${EXIT_CODE} EQUAL 0)
 endif()
 set(NODE_BIN_DIRECTORY "${PROJECT_SOURCE_DIR}/node_modules/.bin")
 
-set(javascript_exec "protoc")
+set(javascript_exec "${PROJECT_SOURCE_DIR}/src/bin/protoc")
```
[the remark this answers is not in the document]
Comment (arsenetar) [reply CMakeLists.txt:117]: The binary is actually already in that location, and the docker build works, other platforms could be an issue as it is a linux amd64 binary.  This is just putting this back in a state that actually works, as without it some systems (like mine) have a system protoc version that is not able to be used for TS/JS generation.  

I would prefer us to not be building the proto definitions manually most of the time anyhow and relying on Actions here to build them as it provides a more controlled environment.

[the remark this answers is not in the document]
Comment (palp) [reply CMakeLists.txt:117]: Yeah, I'm aware it'll work fine in most scenarios, which is why I'm fine with this being merged as is - just wanted to make a note.

File: src/proto/project.proto [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  optional string owner = 1; // The ID of the organization owning the project, empty for default org for user
+}
+
+message GetDeleteProjectRequest {
+  uint64 id = 1;             // ID of the project to request
+  optional string owner = 2; // The ID of the organization owning the project, empty for default org for user
+}
+
+// Message for empty requests / response where no additional data is needed
+message ProjectEmpty {
```
Comment (wbrown) [inline src/proto/project.proto:68]: What I've found useful is to return the data structure being deleted, but with a `deleted` status to communicate that X was deleted.

File: src/proto/project.proto [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+message ProjectAsset {
+  uint64 id = 1;           // ID of the asset
+  string URI = 2;          // The URI to the asset
+  ProjectAssetUse use = 3; // The use of the asset with respect to the project
+}
+
+message Project {
+  uint64 id = 1;                     // ID of the project
```
Comment (wbrown) [inline src/proto/project.proto:29]: suggest an uuid v4 and translating that to a binary lookup on the db/backend side.

**Review by wbrown — COMMENTED** [review]
Some general ideas and feedback.

File: src/proto/project.proto (hunk 3 above)
Comment (arsenetar) [reply src/proto/project.proto:29]: If we use a UUID here, then we should be using across the board for these in my opinion no reason to translate UUID to ID.

File: src/proto/project.proto (hunk 2 above)
Comment (arsenetar) [reply src/proto/project.proto:68]: I don't have any issue with returning it.

**Review by wbrown — APPROVED** [review]

**Pull request merged**